qa/rpc-tests/getmocktime.py

```diff
@@ -0,0 +1,53 @@
+#!/usr/bin/env python3
+# Copyright (c) 2023 The Dogecoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+#
+
+from test_framework.mininode import * #NodeConnCB, NODE_NETWORK, NodeConn, wait_until
+from test_framework.test_framework import BitcoinTestFramework
+from test_framework.util import *
+import time
+
+'''
+GetMockTime -- test getmocktime
+'''
+
+class GetMockTimeTestNode(SingleNodeConnCB):
+    def __init__(self):
+        SingleNodeConnCB.__init__(self)
+
+class GetMockTime(BitcoinTestFramework):
+    def __init__(self):
+        super().__init__()
+        self.setup_clean_chain = True
+        self.num_nodes = 1
+        self.counter = 0
+        self.mocktime = int(time.time())
+        self.is_network_split = False
+
+    def run_test(self):
+        self.test_getmocktime()
+
+    def setup_network(self):
+        # set up full nodes
+        self.nodes = []
+        # Node 0 is going to be our testsubject
+        self.nodes.append(start_node(0, self.options.tmpdir, ["-debug=net", "-debug=mempool", "-peertimeout=999999999"]))
+        self.create_testnode()
+
+    def create_testnode(self, node_idx=0):
+        node = GetMockTimeTestNode()
+        conn = NodeConn('127.0.0.1', p2p_port(node_idx), self.nodes[node_idx], node)
+        node.add_connection(conn)
+        return node
+
+    def mocktime_is_set(self):
+        return self.nodes[0].getmocktime() == 123 # lambda checks if the specific mocktime has been set 
+
+    def test_getmocktime(self):
+        self.nodes[0].setmocktime(123) # set the mocktime
+        assert wait_until(self.mocktime_is_set, timeout=60) # execute the lambda repeatedly for at most 60 seconds
```

Comment:
so i just used this and added it underneath wait_until in mininode.py so it can be used in multiple tests instead of just this one. also added some of those test cases and wanted to test overflows but couldn't seem to get the assert_raise_jsonrpc error to work when setting +1 over max or min for int64_t so i'm just testing that max and min work. lmk if i should add or remove any test cases.